lib/chef/resource/windows_ad_join.rb

```diff
@@ -182,10 +182,10 @@ def node_domain
         #   workgroup.
         #
         def node_workgroup
-          node_workgroup = powershell_out!("(Get-WmiObject Win32_ComputerSystem).Workgroup")
+          node_workgroup = powershell_exec!("(Get-WmiObject Win32_ComputerSystem).Workgroup")
           raise "Failed to check if the system is currently a member of a workgroup" if node_workgroup.error?
 
-          node_workgroup.stdout.downcase.strip
+          node_workgroup.result
```

Comment:
Any reason to remove the `.downcase.strip` here and not in `node_domain`?

Reply:
yeah. workgroup is only ever checked for `empty?` and if one is domain joined, it will return `{}` and break on `downcase.strip`.